enterprise/management/commands/create_enterprise_course_enrollments.py

```diff
@@ -6,119 +6,134 @@
-
-        enterprise_learners = enterprise_customer.enterprise_customer_users.all()
-        for course_id in course_ids:
-            if not self.get_course_details(course_id):
-                LOGGER.warning('Course {course} not found, skipping.'.format(course=course_id))
-                continue
-
-            enrolled_users_count, ent_course_enrollments_count = self.create_enterprise_course_enrollments(
-                course_id,
-                enterprise_learners
-            )
-
-            LOGGER.info(
-                'Created {created} missing EnterpriseCourseEnrollments '
-                'for {enrolled} enterprise learners enrolled in {course}.'.format(
-                    created=ent_course_enrollments_count,
-                    enrolled=enrolled_users_count,
-                    course=course_id
-                )
-            )
-
-    def get_course_details(self, course_id):
-        """
-        Returns course details for the given course or None if the course details
-        could not be retrieved from the LMS courses API.
-
-        Arguments:
-            course_id (string): The course ID.
-
-        Returns:
-            dict: The course details or None if the course could not be retrieved.
-        """
-        try:
-            return CourseApiClient().get_course_details(course_id)
-        except (RequestException, SlumberBaseException, HttpClientError):
-            LOGGER.error('Failed to retrieve course details from LMS API for {course}'.format(course=course_id))
+        enterprise_customer_uuid_filter = options.get('enterprise_customer_uuid')
+        records_created = 0
 
-        return None
+        missing_enrollment_data = self._fetch_course_enrollment_data(
+            enterprise_customer_uuid_filter
+        )
+        for item in missing_enrollment_data:
+            enterprise_customer_uuid = item['enterprise_customer_uuid']
+            user_id = item['user_id']
+            course_run_id = item['course_run_id']
+
+            # pylint: disable=no-member
+            enterprise_customer = EnterpriseCustomer.objects.get(uuid=enterprise_customer_uuid)
+            if enterprise_customer.catalog_contains_course(course_run_id):
+                enterprise_customer_user = EnterpriseCustomerUser.objects.get(
+                    enterprise_customer=enterprise_customer_uuid,
+                    user_id=user_id
+                )
 
-    def create_enterprise_course_enrollments(self, course_id, enterprise_learners):
+                # pylint: disable=no-member
+                __, created = EnterpriseCourseEnrollment.objects.get_or_create(
+                    enterprise_customer_user=enterprise_customer_user,
+                    course_id=course_run_id
+                )
+                if created:
+                    records_created += 1
+                    LOGGER.info(
+                        'EnterpriseCustomer [%s] - User [%s] - CourseRun [%s]',
```

Comment:
consider prepending this for clarity: 'Created EnterpriseCourseEnrollment for: '